src/UI/Component/SignalGenerator.php

```diff
@@ -0,0 +1,18 @@
+<?php
+namespace ILIAS\UI\Component;
+
+/**
+ * Interface SignalGenerator
```

Comment:
At some place we should describe, what a signal is and how it works. Maybe this could be the place.